.github/workflows/bench.yml

```diff
@@ -16,7 +16,18 @@ name: bench
 jobs:
   codspeed:
     runs-on: ubuntu-latest
+    # Set the CODSPEED_TOKEN as an environment variable at the job level
+    env:
+      CODSPEED_TOKEN: ${{ secrets.CODSPEED_TOKEN }}
     steps:
+      # Check token early and exit if not available
+      - name: Check CODSPEED_TOKEN availability
+        run: |
+          if [ -z "$CODSPEED_TOKEN" ]; then
+            echo "::notice::CODSPEED_TOKEN is not set, skipping benchmarks"
+            echo "This is expected for forks. Benchmarks only run in the upstream repository."
+            exit 0
```

Comment:
the check fails in latest CI run,so maybe exit 0 doesn't work here like i expected. wdyt @dhyaniarun1993 ?